annotated.go

```diff
@@ -144,6 +144,82 @@ type paramTagsAnnotation struct {
 }
 
 var _ Annotation = paramTagsAnnotation{}
+var (
+	errTagSyntaxSpace            = errors.New(` multiple tags are not separated by space`)
+	errTagKeySyntax              = errors.New(" tag key is invalid, Use group, name or optional as tag keys")
+	errTagValueSyntaxQuote       = errors.New(`tag value should start with double quote. i.e. key:"value" `)
+	errTagValueSyntaxEndingQuote = errors.New(`tag value should end in double quote. i.e. key:"value" `)
+)
+
+// Given a tag with key, value pair, they must be separaed by a space.
+// Eg: `group:"some" optional:"true"`
+// return an error if in the case of multiple tags, they aren't separated
+// space. Return nil on success.
+func verifyTagsSpaceSeparated(numTags int, tag string) error {
+	if numTags > 0 && tag != "" && tag[0] != ' ' {
+		return errTagSyntaxSpace
+	}
+	return nil
+}
+
+// Given a value from a tag verify that it is contained between double quotes
+// Return an error if the invalid quote format is used for tag value.
+// On sucess return the remaining tag with nil error
+func verifyValueQuote(value string) (string, error) {
+	// starting quote should be a double quote
```

Comment:
Could we just check if the value string is prefixed and suffixed with double quotes. Could the step to validate tag values is within quotes be replaced by counting the total number of double quotes and ensuring that is exactly 2?

Reply:
The total number of double quotes can be more than 2, as it's valid for a struct string to have quotes in it. 
`key:"val/"ue"` would be a valid tag with more than 2 double quotes. Hence, I don't think it'd be possible unless I misunderstood?

Reply:
Ah understood that this approach does not work because value actually contains the rest of the tag strings.